src/components/ShoppingList/index.tsx

```diff
@@ -1,18 +1,40 @@
-import { Button, Flex } from '@chakra-ui/react';
+import { Flex, IconButton, Text, Badge } from '@chakra-ui/react';
+import { useState } from 'react';
 import { MdOutlineShoppingCart } from 'react-icons/md';
 
 export const ShoppingList = () => {
+  const [count] = useState(3);
+
   return (
-    <Flex h="42px" w="42">
-      <Button
-        h="full"
-        w="full"
-        backgroundColor="#F9A109"
+    <Flex w="full" justify="center">
+      <IconButton
+        position="relative"
+        w="42px"
+        h="42px"
         borderRadius="50%"
-        color={'white'}
-      >
-        <MdOutlineShoppingCart fontSize={20} />
-      </Button>
+        color="white"
+        bg="#F9A109"
+        _hover={{ bg: '#FF8E22' }}
+        _active={{ bg: '#D5771D' }}
+        aria-label={'Shopping cart'}
+        icon={
+          <>
+            <MdOutlineShoppingCart fontSize={22} />
+
+            <Badge
```

Comment:
The chakra IconButton doesn't have a `buttonBadge` prop, to add the badge inside the "clickable" button, I put it inside the icon prop.

Reply:
I'm not expert with chakra UI, but I believe is the right way to do that.